vhost/src/vhost_user/gpu_backend_req.rs

```diff
@@ -55,6 +58,21 @@ impl BackendInternal {
         Ok(())
     }
 
+    fn send_message_with_payload<T: ByteValued>(
+        &mut self,
+        request: GpuBackendReq,
+        body: &T,
+        data: &[u8],
+        fds: Option<&[RawFd]>,
+    ) -> vhost_user::Result<()> {
+        self.check_state()?;
+
+        let len = mem::size_of::<T>();
```

Comment:
You also need to account for the length of the payload.
```suggestion
        let len = mem::size_of::<T>() + data.len();
```